patch-package
--- a/node_modules/interactjs/index.d.ts
+++ b/node_modules/interactjs/index.d.ts
@@ -29,10 +29,20 @@ declare namespace interact {
     height: number
   }
 
+  interface Rect3 {
+    width?: number
+    height?: number
+  }
+
   interface SnapFunction {
     ( x: number, y: number ) : SnapPosition
   }
 
+  interface RestrictSizeOption {
+      min?: Rect3
+      max?: Rect3
+  }
+
   type SnapTarget = SnapPosition | SnapFunction
   type SnapOptions = {
     targets?: SnapTarget[]
@@ -97,6 +107,7 @@ declare namespace interact {
   }
 
   interface ResizableOptions {
+    enabled?: boolean
     max?: number
     maxPerElement?: number
     manualStart?: boolean
@@ -111,6 +122,7 @@ declare namespace interact {
     axis?: 'x' | 'y'
     //
     invert?: 'none' | 'negate' | 'reposition'
+    restrictSize?: RestrictSizeOption
     squareResize?: boolean
     onstart?: Listener
     onmove?: Listener
@@ -192,6 +204,7 @@ declare namespace interact {
     dragLeave?: DOMElement
     // resize
     axes: Position
+    rect: any
     // gestureend
     distance: number
     angle: number
