patch-package
--- a/node_modules/ngx-treeview/src/dropdown-menu.directive.d.ts
+++ b/node_modules/ngx-treeview/src/dropdown-menu.directive.d.ts
@@ -1,5 +1,5 @@
 import { DropdownDirective } from './dropdown.directive';
 export declare class DropdownMenuDirective {
-    private dropdown;
+    public dropdown;
     constructor(dropdown: DropdownDirective);
 }
--- a/node_modules/ngx-treeview/src/dropdown-toggle.directive.d.ts
+++ b/node_modules/ngx-treeview/src/dropdown-toggle.directive.d.ts
@@ -1,6 +1,6 @@
 import { ElementRef } from '@angular/core';
 import { DropdownDirective } from './dropdown.directive';
 export declare class DropdownToggleDirective {
-    private dropdown;
+    public dropdown;
     constructor(dropdown: DropdownDirective, elementRef: ElementRef);
 }
