diff --git a/node_modules/webstomp-client/dist/webstomp.js b/node_modules/webstomp-client/dist/webstomp.js
index 63e2d69..e4f706b 100644
--- a/node_modules/webstomp-client/dist/webstomp.js
+++ b/node_modules/webstomp-client/dist/webstomp.js
@@ -422,7 +422,7 @@
                               _this.debug('connected to server ' + frame.headers.server);
                               _this.connected = true;
                               _this.version = frame.headers.version;
-                              _this._setupHeartbeat(frame.headers);
+                              _this._setupHeartbeat(frame.headers, errorCallback);
                               if (connectCallback) connectCallback(frame);
                               break;
                           // [MESSAGE Frame](http://stomp.github.com/stomp-specification-1.1.html#MESSAGE)
@@ -718,7 +718,7 @@
 
       }, {
           key: '_setupHeartbeat',
-          value: function _setupHeartbeat(headers) {
+          value: function _setupHeartbeat(headers, errorCallback) {
               var _this2 = this;
 
               if (this.version !== VERSIONS.V1_1 && this.version !== VERSIONS.V1_2) return;
@@ -751,6 +751,7 @@
                       // We wait twice the TTL to be flexible on window's setInterval calls
                       if (delta > _ttl * 2) {
                           _this2.debug('did not receive server activity for the last ' + delta + 'ms');
+                          if (errorCallback) errorCallback(`did not receive server activity for the last ${delta}ms`);
                           _this2.ws.close();
                       }
                   }, _ttl);
